src/hats/catalog/dataset/table_properties.py

```diff
@@ -0,0 +1,163 @@
+        "join_column",
+        "contains_leaf_files",
+    ],
+    CatalogType.INDEX: ["primary_catalog", "indexing_column"],
+    CatalogType.MARGIN: ["primary_catalog", "margin_threshold"],
+}
+
+
+class TableProperties(BaseModel):
+    """Container class for catalog metadata"""
+
+    catalog_name: str = Field(alias="obs_collection")
+    catalog_type: CatalogType = Field(alias="dataproduct_type")
+    total_rows: int = Field(alias="hats_nrows")
+
+    ra_column: Optional[str] = Field(default=None, alias="hats_col_j2000_ra")
+    dec_column: Optional[str] = Field(default=None, alias="hats_col_j2000_dec")
+
+    primary_catalog: Optional[str] = Field(default=None, alias="hats_primary_table_url")
+    """Reference to object catalog. Relevant for nested, margin, association, and index"""
+
+    margin_threshold: Optional[float] = Field(default=None, alias="hats_margin_threshold")
+    """Threshold of the pixel boundary, expressed in arcseconds."""
+
+    primary_column: Optional[str] = Field(default=None, alias="hats_col_assn_primary")
+    """Column name in the primary (left) side of join"""
+
+    primary_column_association: Optional[str] = Field(default=None, alias="hats_col_assn_primary_assn")
+    """Column name in the association table that matches the primary (left) side of join"""
+
+    join_catalog: Optional[str] = Field(default=None, alias="hats_assn_join_table_url")
+    """Catalog name for the joining (right) side of association"""
+
+    join_column: Optional[str] = Field(default=None, alias="hats_col_assn_join")
+    """Column name in the joining (right) side of join"""
+
+    join_column_association: Optional[str] = Field(default=None, alias="hats_col_assn_join_assn")
+    """Column name in the association table that matches the joining (right) side of join"""
+
+    contains_leaf_files: Optional[bool] = Field(default=None, alias="hats_assn_leaf_files")
+    """Whether or not the association catalog contains leaf parquet files"""
+
+    indexing_column: Optional[str] = Field(default=None, alias="hats_index_column")
+    """Column that we provide an index over"""
+
+    extra_columns: Optional[List[str]] = Field(default=None, alias="hats_index_extra_column")
+    """Any additional payload columns included in index"""
+
+    ## Allow any extra keyword args to be stored on the properties object.
+    model_config = ConfigDict(extra="allow", populate_by_name=True, use_enum_values=True)
+
+    @field_validator("extra_columns", mode="before")
+    @classmethod
+    def space_delimited_list(cls, str_value: str) -> List[str]:
+        """Convert a space-delimited list string into a python list of strings."""
+        if isinstance(str_value, str):
+            if str_value:
+                str_value = str_value.strip()
+            if str_value:
+                return str_value.split(" ")
+        return str_value
+
+    @field_serializer("extra_columns")
+    def serialize_space_delimited_list(self, str_list) -> str:
```

Comment:
```suggestion
    def serialize_as_space_delimited_list(self, str_list: list[str]) -> str:
```

Reply:
Iterable will make it useful for other future fields, so thank you!